From 64afa93fbbca3efc9f4695e86e46e3c4ff03f4d1 Mon Sep 17 00:00:00 2001
From: Markus Volk <f_l_k@t-online.de>
Date: Thu, 20 May 2021 16:19:10 +0200
Subject: [PATCH] meson: make ELF-TLS optional

Signed-off-by: Markus Volk <f_l_k@t-online.de>
---
 meson.build       | 3 ++-
 meson_options.txt | 8 ++++++++
 2 files changed, 10 insertions(+), 1 deletions(-)

diff --git a/meson.build b/meson.build
index e1e94e71049..d9f27b5ec76 100644
--- a/meson.build
+++ b/meson.build
@@ -449,7 +449,8 @@ endif
 use_elf_tls = false
 if (not ['freebsd', 'openbsd', 'haiku'].contains(host_machine.system()) and
     (not with_platform_android or get_option('platform-sdk-version') >= 29) and
-    (not with_platform_windows or not with_shared_glapi))
+    (not with_platform_windows or not with_shared_glapi) and
+    get_option('elf-tls'))
   pre_args += '-DUSE_ELF_TLS'
   use_elf_tls = true
 endif
diff --git a/meson_options.txt b/meson_options.txt
index a7030aba31e..2dfffac3d2a 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -421,6 +421,14 @@ option(
   value : true,
   description : 'Enable direct rendering in GLX and EGL for DRI',
 )
+
+option(
+  'elf-tls',
+  type : 'boolean',
+  value : true,
+  description : 'Enable TLS support in ELF',
+)
+
 option(
   'prefer-iris',
   type : 'boolean',
--
2.25.1
